src/rfxencode_diff_rlgr1.c

```diff
@@ -0,0 +1,252 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol client.
+ * RemoteFX Codec Library - RLGR
+ *
+ * Copyright 2011 Vic Lee
+ * Copyright 2016-2017 Jay Sorg <jay.sorg@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This implementation of RLGR refers to
+ * [MS-RDPRFX] 3.1.8.1.7.3 RLGR1/RLGR3 Pseudocode
+ */
+
+#if defined(HAVE_CONFIG_H)
+#include <config_ac.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "rfxcommon.h"
+
+#define PIXELS_IN_TILE 4096
+
+/* Constants used within the RLGR1/RLGR3 algorithm */
+#define KPMAX   (80)  /* max value for kp or krp */
+#define LSGR    (3)   /* shift count to convert kp to k */
+#define UP_GR   (4)   /* increase in kp after a zero run in RL mode */
+#define DN_GR   (6)   /* decrease in kp after a nonzero symbol in RL mode */
+#define UQ_GR   (3)   /* increase in kp after nonzero symbol in GR mode */
+#define DQ_GR   (3)   /* decrease in kp after zero symbol in GR mode */
```

Comment:
Those constants are used in two places. It would be better to have them in a header. The names are too short for my taste. How about KP_MAX, KP_SHIFT, KP_RL_INC, KP_RL_DEC, KP_GR_INC, and KP_GR_DEC? The use of "GR" for RL mode is misleading.